api/client/container/update.go

```diff
@@ -0,0 +1,150 @@
+package container
+
+import (
+	"fmt"
+	"strings"
+
+	"golang.org/x/net/context"
+
+	"github.com/docker/docker/api/client"
+	"github.com/docker/docker/cli"
+	runconfigopts "github.com/docker/docker/runconfig/opts"
+	containertypes "github.com/docker/engine-api/types/container"
+	"github.com/docker/go-units"
+	"github.com/spf13/cobra"
+)
+
+type updateOptions struct {
+	blkioWeight       *uint16
+	cpuPeriod         *int64
+	cpuQuota          *int64
+	cpusetCpus        *string
+	cpusetMems        *string
+	cpuShares         *int64
+	memoryString      *string
+	memoryReservation *string
+	memorySwap        *string
+	kernelMemory      *string
+	restartPolicy     *string
+
+	nFlag int
+
+	containers []string
+}
+
+// NewUpdateCommand creats a new cobra.Command for `docker update`
+func NewUpdateCommand(dockerCli *client.DockerCli) *cobra.Command {
+	var opts updateOptions
+
+	cmd := &cobra.Command{
+		Use:   "update [OPTIONS] CONTAINER [CONTAINER...]",
+		Short: "Update configuration of one or more containers",
+		Args:  cli.RequiresMinArgs(1),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			opts.containers = args
+			opts.nFlag = cmd.Flags().NFlag()
+			return runUpdate(dockerCli, &opts)
+		},
+	}
+	cmd.SetFlagErrorFunc(flagErrorFunc)
+
+	flags := cmd.Flags()
+	opts.blkioWeight = flags.Uint16("blkio-weight", 0, "Block IO (relative weight), between 10 and 1000")
```

Comment:
Use `flags.Uint16Var` to avoid pointers. Take `api/client/container/run.go` as reference. Thanks!